rcl_lifecycle/include/rcl_lifecycle/transition_map.h

```diff
@@ -25,24 +25,87 @@ extern "C"
 {
 #endif
 
+/// Initialize a rcl_lifecycle_state_init.
+/*
+ * Should be called to get a null rcl_lifecycle_transition_map_t before passing to
+ * rcl_lifecycle_register_state() or rcl_lifecycle_register_transition().
+ *
+ * \return rcl_lifecycle_transition_map_t a initilized struct
+ */
 RCL_LIFECYCLE_PUBLIC
 RCL_WARN_UNUSED
 rcl_lifecycle_transition_map_t
 rcl_lifecycle_get_zero_initialized_transition_map();
 
+/// Check if a transition map is active using a rcl_lifecycle_state_machine_t.
+/*
+ * The function checks if the transition map is initialized. It returns `RCL_RET_OK`
+ * if the transition map is initialized successfully or `RCL_RET_ERROR` if the transition
+ * map is not initialized.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | No
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[in] transition_map pointer to the transition map struct to check
+ * \return `RCL_RET_OK` if the transition map is initialized successfully, or
+ * \return `RCL_RET_ERROR` if the transition map is not initialized.
+ */
 RCL_LIFECYCLE_PUBLIC
 RCL_WARN_UNUSED
 rcl_ret_t
 rcl_lifecycle_transition_map_is_initialized(
   const rcl_lifecycle_transition_map_t * transition_map);
 
+/// Finalize a rcl_lifecycle_transition_map_t.
+/*
+ * Calling this will set the rcl_lifecycle_transition_map_t struct into the an unitialized state that is
+ * functionally the same as before rcl_lifecycle_register_state or
+ * rcl_lifecycle_register_transition was called.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[inout] transition_map struct to be deinitialized
+ * \param[in] allocator a valid allocator used to deinitialized the state machine
+ * \return `RCL_RET_OK` if the state was deinitialized successfully, or
+ * \return `RCL_RET_ERROR` if an unspecified error occurs.
+ */
 RCL_LIFECYCLE_PUBLIC
 RCL_WARN_UNUSED
 rcl_ret_t
 rcl_lifecycle_transition_map_fini(
   rcl_lifecycle_transition_map_t * transition_map,
   const rcl_allocator_t * allocator);
 
+/// Regiter a state
+/*
+ * This function registers a new state in the transition map.
+ *
+ * <hr>
+ * Attribute          | Adherence
+ * ------------------ | -------------
+ * Allocates Memory   | Yes
+ * Thread-Safe        | No
+ * Uses Atomics       | No
+ * Lock-Free          | Yes
+ *
+ * \param[in] map
+ * \param[in] state the state to register
+ * \param[in] allocator a valid allocator used to deinitialized the state machine
```

Comment:
I think it reallocates it, I wouldn't use deinitialize in this context, maybe "used to register" or "used to reallocate"